### PR TITLE
Fixes Brave Ads fail to migrate confirmations.json if upgrading from builds prior to 1.12.2 if ads were switched off

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/12260

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/12260

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by yachtcaptain23 — APPROVED** [review]

File: components/brave_ads/browser/ads_service_impl.cc [hunk 1]
```diff
@@ -573,8 +573,19 @@ bool MigrateConfirmationsStateOnFileTaskRunner(
       rewards_service_base_path.AppendASCII("confirmations.json");
 
   if (base::PathExists(legacy_confirmations_state_path)) {
+    const base::FilePath ads_service_base_path = path.AppendASCII("ad_service");
```
Comment (emerick) [inline components/brave_ads/browser/ads_service_impl.cc:576]: Should this be "ads_service" or is that part of the fix?

Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.cc]: :-) good spot as I just spotted that during my final testing. Fixed

**Review by emerick — APPROVED** [review]
LGTM

**Review by moritzhaller — APPROVED** [review]
lgtm!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.7 x64` using the following build:

```
Brave | 1.18.49 Chromium: 87.0.4280.49 (Official Build) nightly (x86_64)
-- | --
Revision | f77f85899646b42a1d3c8ff36794e00becab9171-refs/branch-heads/4280@{#1115}
OS | macOS Version 10.15.7 (Build 19H2)
```

Installed `1.14.19 CR: 84.0.4147.105`, enabled rewards and disabled ads via `brave://rewards`. Once disabled, updated to `1.18.44 CR: 87.0.4280.49` and ensured ads were being initialized as per the following:

```
[4091:775:1111/152940.554844:VERBOSE1:ads_impl.cc(111)] Initializing ads
[4091:775:1111/152940.603853:WARNING:CONSOLE(1)] "Could not find style data in module named br-shared-style", source: chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js (1)
[4091:775:1111/152940.607062:VERBOSE1:page_classifier.cc(154)] Successfully loaded emgmepnebbddgnkhfmhdhmjifkglkamo page classification user model
[4091:775:1111/152940.841043:VERBOSE1:page_classifier.cc(163)] Successfully initialized emgmepnebbddgnkhfmhdhmjifkglkamo page classification user model
[4091:775:1111/152940.841211:VERBOSE1:database_migration.cc(53)] Migrated database from version 1 to version 5
[4091:775:1111/152940.845077:VERBOSE3:client.cc(466)] Loading client state
[4091:775:1111/152940.845548:VERBOSE3:client.cc(492)] Successfully loaded client state
[4091:775:1111/152940.845586:VERBOSE3:confirmations.cc(208)] Loading confirmations state
[4091:775:1111/152940.845901:VERBOSE3:confirmations.cc(218)] Confirmations state does not exist, creating default state
[4091:775:1111/152940.845969:VERBOSE3:ad_notifications.cc(378)] Loading ad notifications state
[4091:775:1111/152940.857026:VERBOSE3:ad_notifications.cc(404)] Successfully loaded ad notifications state
[4091:775:1111/152940.857069:VERBOSE3:conversions.cc(437)] Loading conversions state
[4091:775:1111/152940.857356:VERBOSE3:conversions.cc(463)] Successfully loaded conversions state
[4091:775:1111/152940.857385:VERBOSE1:ads_impl.cc(441)] Successfully initialized ads
```

Also ensured that after upgrading to `1.18.44 CR: 87.0.4280.49`, ads were working/being displayed as per the following:

```
[4091:775:1111/153049.924128:VERBOSE1:ad_notification_serving.cc(95)] Ad notification delivered:
  uuid: e8af2f5a-20a4-4bf0-b031-2478a7bc741f
  creativeInstanceId: c0084199-d950-44bc-af98-7f6a379d238a
  creativeSetId: 6256002e-a7dd-4bb4-8853-0ae7bb9cb833
  campaignId: 42458d7e-eba5-4537-823d-ffca381d2b3b
  category: technology & computing
  title: Nexo’s Savings Account
  body: Earn up to 8% APY on your BTC & ETH, with Nexo!
  targetUrl: https://nexo.io/earn-fiat?utm_source=brave&utm_medium=fixed&utm_campaign=brave_push_earnonfiat_oct20&utm_term=earnonfiat
```

Ensured that both `Ad notifications received this month` & `Estimated pending rewards` were being updated.

**Pull request merged**